### PR TITLE
UI enhancement

### PR DESCRIPTION
**Pull request by purushottamyadavbattula**
Added

- Caching of stories in home vie model
- Paging of stories
- Time delay for fetching stories from API
- Animated loading button

### CODE REVIEW

File: lib/app/app.dart [hunk 1]
```diff
@@ -11,7 +12,8 @@ import 'package:stacked_services/stacked_services.dart';
 ], dependencies: [
   LazySingleton(classType: NavigationService),
   LazySingleton(classType: APIService),
-  LazySingleton(classType: CommonServices)
+  LazySingleton(classType: CommonServices),
+  LazySingleton(classType: HomeViewModel)
```
Comment (priyupadhyay) [inline lib/app/app.dart]: Remove `HomeViewModel` from here.

File: lib/ui/views/home/home_view.dart [hunk 2]
```diff
@@ -23,34 +26,54 @@ class HomeView extends StatelessWidget {
...
-                      },
-                      title: Text(model.stories[index].title,
-                          style: h4.copyWith(color: TextColorDark)),
-                      subtitle: Text(model.stories[index].by),
-                    );
+                child: NotificationListener<ScrollNotification>(
+                  onNotification: (ScrollNotification scrollInfo) {
+                    if (scrollInfo.metrics.pixels ==
+                        scrollInfo.metrics.maxScrollExtent) {
+                      model.getNext();
```
Comment (priyupadhyay) [inline lib/ui/views/home/home_view.dart:53]: add a `loading more stories` text at the end of the list.

Comment (purushottamyadavbattula) [reply lib/ui/views/home/home_view.dart:53]: This for pagination, we can keep it since it gets data from the cache

File: lib/app/app.dart (hunk 1 above)
Comment (purushottamyadavbattula) [reply lib/app/app.dart]: sure

**Review by priyupadhyay — APPROVED** [review]

**Pull request merged**